AEFeinstein/Super-2024-Swadge-FW
# Emulator Improvements and OSX Build

### Description

A lot of emulator functionality, some of it directly related to making the OSX bundle work and some of it unrelated.

* NVS defaults to `{"storage":{"test":1}}` so that the factory test isn't used by default
* NVS attempts to open `./nvs.json` first, but if that fails it will try `~/Library/Preferences/org.magfest.SwadgeEmulator/nvs.json` on OSX, and `~/.swadge_emulator_nvs.json` on Linux. This was done because OSX starts bundle apps in `/` so the emulator can never actually write to `./nvs.json`.
* The random seed can now be set with a command-line arg, and is also automatically saved when using `--record` and used with `--playback` so you can get consistent results even if a mode uses `esp_random()`
* Record the screen to a GIF with F12
* Pause the emulator with F10 and single frame step with F9.
* Show a built-in FPS counter with F5
* Various time-related options, such as forcing a fake consistent timer, fuzzing frame lengths, and forcing a fake framerate
* I couldn't help myself and I added a console to the emulator, openable with F4. I think I'll eventually need this to make MIDI on the emulator properly configurable
* Added an Emulator User Guide to the docs that just covers functionality and command-line args

### Test Instructions

Run the emulator with various command-line args, use the F-keys to do random stuff, record some GIFS. Mess with the time options, see if you can break anything.

### Readiness Checklist
